src/nn/wasm/src/bind/src/binding.cpp

```diff
@@ -0,0 +1,342 @@
+                                   const intptr_t biasData, 
+                                   const RuntimeShape& outputShape, 
+                                   intptr_t outputData) {
+    optimized_ops::DepthwiseConv(op_params,
+                                 inputShape, (const float*)inputData, 
+                                 filterShape, (const float*)filterData, 
+                                 biasShape, (const float*)biasData, 
+                                 outputShape, (float*)outputData);
+  }
+
+  bool convFloat32Wrapper(const ConvParams& op_params, 
+                          const RuntimeShape& inputShape, 
+                          const intptr_t inputData, 
+                          const RuntimeShape& filterShape, 
+                          const intptr_t filterData, 
+                          const RuntimeShape& biasShape, 
+                          const intptr_t biasData, 
+                          const RuntimeShape& outputShape, 
+                          intptr_t outputData,
+                          const RuntimeShape& im2colShape, 
+                          intptr_t im2colData) {
+    optimized_ops::Conv(op_params, 
+                        inputShape, (const float*)inputData, 
+                        filterShape, (const float*)filterData, 
+                        biasShape, (const float*)biasData, 
+                        outputShape, (float*)outputData, 
+                        im2colShape, (float*)im2colData);
+  }
+
+  bool averagePoolFloat32Wrapper(const PoolParams op_params,
+                                 const RuntimeShape& inputShape, 
+                                 const intptr_t inputData, 
+                                 const RuntimeShape& outputShape, 
+                                 intptr_t outputData) {
+    optimized_ops::AveragePool(op_params,
+                               inputShape, (const float*)inputData,
+                               outputShape, (float*)outputData);
+  }
+
+  bool maxPoolFloat32Wrapper(const PoolParams op_params,
+                             const RuntimeShape& inputShape, 
+                             const intptr_t inputData, 
+                             const RuntimeShape& outputShape, 
+                             intptr_t outputData) {
+    optimized_ops::MaxPool(op_params,
+                           inputShape, (const float*)inputData,
+                           outputShape, (float*)outputData);
+  }
+
+  bool softmaxFloat32Wrapper(const SoftmaxParams op_params,
+                             const RuntimeShape& inputShape, 
+                             const intptr_t inputData, 
+                             const RuntimeShape& outputShape, 
+                             intptr_t outputData) {
+    optimized_ops::Softmax(op_params, inputShape, (const float*)inputData,
+                           outputShape, (float*)outputData);
+  }
+
+  bool reshapeGenericWrapper(const binding_utils::ReshapeParams op_params,
```

Comment:
Why not compute the size by `inputShape`?

Reply:
I calculate it by `inputShape` in js, just to make all the init steps do in the js code. 
```js
if (utils.isTensor(input.type)) {
    size_count = utils.sizeOfTensorData(input.type, input.runtimeshape.dims);
} else {
    size_count = utils.sizeOfScalarData(input.type);
}
```
Move it to C++ code?

Reply:
In this way, I suppose you don't need introduce `ReshapeParams`. It would make the binding code simpler, won't it?